common-extensions/src/main/java/com/twocoders/util/commonextensions/ContextExtensions.kt

```diff
@@ -0,0 +1,266 @@
+    get() = applicationContext.getSystemService(TELEPHONY_SERVICE) as? TelephonyManager
+
+inline val Context.uiModeManager: UiModeManager?
+    get() = applicationContext.getSystemService(UI_MODE_SERVICE) as? UiModeManager
+
+inline val Context.downloadManager: DownloadManager?
+    get() = applicationContext.getSystemService(DOWNLOAD_SERVICE) as? DownloadManager
+
+inline val Context.batteryManager: BatteryManager?
+    get() = applicationContext.getSystemService(BATTERY_SERVICE) as? BatteryManager
+
+inline val Context.hardwarePropertiesManager: HardwarePropertiesManager?
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    get() = applicationContext.getSystemService(HARDWARE_PROPERTIES_SERVICE) as? HardwarePropertiesManager
+
+fun Context.isNetworkAvailable(): Boolean = (connectivityManager).run {
+    if (isMarshmallow()) {
+        return this?.getNetworkCapabilities(activeNetwork)?.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) ?: false
+    } else {
+        (@Suppress("DEPRECATION")
+        return this?.activeNetworkInfo?.isConnected ?: false)
+    }
+}
+
+fun Context.isNetworkUnavailable() = !isNetworkAvailable()
+
+fun Context.scanForActivity(): Activity? = when (this) {
+    is Activity -> this
+    is ContextWrapper -> baseContext.scanForActivity()
+    else -> null
+}
+
+fun Context.getAnimationDrawable(@DrawableRes drawableResId: Int): AnimationDrawable? = try {
+    getDrawable(drawableResId = drawableResId) as? AnimationDrawable
+} catch (oom: OutOfMemoryError) {
+    logd("Failed to load the AnimationDrawable:", oom)
+    null
+}
+
+fun Context.getBitmapFromDrawable(@DrawableRes drawableResId: Int, @ColorInt tintColor: Int = UNDEFINED): Bitmap {
+    val drawable = getDrawable(drawableResId = drawableResId) ?: throw Resources.NotFoundException("The given resource Id does not exist :/")
+    return when (drawable) {
+        is BitmapDrawable -> drawable.bitmap
+        is VectorDrawable -> {
+            Bitmap.createBitmap(drawable.intrinsicWidth, drawable.intrinsicHeight, Bitmap.Config.ARGB_8888).apply {
+                val canvas = Canvas(this)
+                drawable.setBounds(0, 0, canvas.width, canvas.height)
+                if (tintColor != UNDEFINED) drawable.setTint(tintColor)
+                drawable.draw(canvas)
+            }
+        }
+        else -> throw IllegalArgumentException("Unsupported drawable type...")
+    }
+}
+
+@SuppressLint("UseCompatLoadingForDrawables")
+// For API 21 and up the ContextCompat.getDrawable() wrapper just call the same method Context.getDrawable()
+fun Context.getDrawable(@DrawableRes drawableResId: Int, theme: Resources.Theme = getTheme()): Drawable? =
+    resources.getDrawable(drawableResId, theme)
+
+@ColorInt
+fun Context.getColorInt(@ColorRes colorResId: Int, theme: Resources.Theme = getTheme()): Int =
+    if (isMarshmallow()) resources.getColor(colorResId, theme) else ContextCompat.getColor(this, colorResId)
```

Comment:
Nie je ulohou `ContextCompat` riesit prave taketo checky?

Reply:
jj, je. Ale proč máš ty jako developer myslet na to, že něco owrapovali přes ContextCompat a máš volat jeho namísto klasické metody na kontextu? tato extension to jen zaobalí aniž by jsi musel rozmýšlet nad tím, zda máš volat `context.getColorStateList()` nebo `ContextCompat.getColorStateList()`;)

Reply:
to je v poriadku, ale preco to potom nie je 
```
@ColorInt
fun Context.getColorInt(@ColorRes colorResId: Int): Int = ContextCompat.getColor(this, colorResId)
```

Reply:
aby se ušetřilo jedno volání metody navýše a aby až bude min API toho projektu vetší než `Marshmallow`, tak aby se celý ten compat mohl poslat do kytek. Se zvednutím API je to díky té `isXY` extension velmi snadno dohledatelné.